python/pyarrow/tests/test_cuda.py

```diff
@@ -57,6 +57,18 @@ def test_Context():
     assert global_context.device_number == 0
     assert global_context1.device_number == cuda.Context.get_num_devices() - 1
 
+    mm = global_context.memory_manager()
+    assert not mm.is_cpu
+    assert "<pyarrow.MemoryManager device: CudaDevice" in repr(mm)
+
+    dev = global_context.device()
+    assert dev == mm.device
+
+    assert not dev.is_cpu
+    assert dev.device_id == 0
+    assert dev.device_type == pa.DeviceAllocationType.CUDA
+    assert "<pyarrow.Device: CudaDevice" in repr(dev)
```

Comment:
nit: IMO asserting on repr output is not necessary. We already check the `device_type`.